### DISCUSSION TITLE
Enabling RSS feed in footer

### DISCUSSION THREAD

**Discussion by ppipada** [razonyang/hugo-theme-bootstrap#441] [Q&A] [answered]
Hi,

How can I enable the rss feed icon in footer of each page and point it to the single main page index.xml? 

I am not using the "author" card in the sidebar. If I use it, I can get it working there, but I am not able to do it in the footer. Any documentation link would also help.

Thanks

**Comment by chrisrummel** [comment, marked as answer]
Here is how I did that:
I copied themes/hugo-theme-bootstrap/layouts/partials/helpers/social-links.html to layouts/partials/helpers/social-links.html so I have a local copy of that file I can change. Make sure to have an eye on possible future changes of that file so you can apply changes.

What I changed:
```
$ diff themes/hugo-theme-bootstrap/layouts/partials/helpers/social-links.html layouts/partials/helpers/social-links.html
53c53
<     <a class="nav-link social-link" target="_blank" href="{{ .Permalink }}" title="RSS" rel="noopener noreferrer">
---
>     <a class="nav-link social-link" target="_blank" href="/index.xml" title="RSS" rel="noopener noreferrer">
```

**Reply by ppipada** [reply]
Thanks a lot !!!

Only additional thing I needed to do was add `page = ["HTML", "RSS"]` in outputs section of config.toml.
Just having home output params in config doesnt help. This would be most probably because social-links.html is getting the page context and .OutputFormats would be empty in page context if above config is not present.

**Comment by razonyang** [comment]
I don't know much about RSS, should we always put the same RSS link `index.xml` on all pages?
@ppipada @chrisrummel

**Reply by chrisrummel** [reply]
In cases of tags or categories it can be useful to be able to subscribe to just a specific RSS-feed. I would leave it as it is, or make it configurable - that is always the best.

**Reply by ppipada** [reply]
Generally RSS subscription would be at head, section or taxonomy based. I dont think any one would use single page based subscription (this is possible in todays model). 
If we could make these head/taxonomy(category/tag) as top level configurable, it would be great.

**Reply by razonyang** [reply]
@ppipada What does the `head` mean?

**Reply by ppipada** [reply]
By head, I meant the landing home page e.g: /index.html. In Hugo terms I think it is PageKind == home.
Sorry for the confusion of terminology.

**Reply by razonyang** [reply]
I see, thanks for the explanation.

**Comment by razonyang** [comment]
I just did research about RSS, and tested it locally, all pages will be shown on the RSS reader if users subscribed the `home` page(`index.xml`).

I have some thoughts to share with you.

## 1. Dynamic RSS social link on the footer

- Taxonomy's RSS links are preferred, such as `categories`, `tags`,  `series` and their terms(`categories/foobar`).
- Fallback to the `home` RSS links for other pages.

## 2. Additional RSS button

Unlike the former, the footer's RSS link is static that linked to `index.xml` all the time. And add an additional RSS button with the same rules of the former on the `setting` panel or `top app bar`.

Please feel free to share your ideas about it.

**Reply by ppipada** [reply]
In addition to 1, if we can give a configuration flag that can fall to link `home` index.xml all the time, it would be great.
Adding a separate RSS button in `settings` or `top app bar` as mentioned in 2 would feel unintuitive and out of place to me.

**Reply by razonyang** [reply]
@ppipada Sounds better. What do you think? @chrisrummel

**Reply by chrisrummel** [reply]
To me it woud make sense to offer RSS feeds like this:

- /index.html => full RSS feed /index.xml
- /categories => category RSS feed
- /tags => tag RSS feed
- /series => series RSS feed
- pages => no feed link or link to /index.xml

If the site maintainer really means it, he/she can create a dedicated feed-page listing all the different feeds that are offered and can be subscribed to.

**Comment by razonyang** [comment]
@ppipada @chrisrummel I created a PR #445 for this, you can preview it on https://deploy-preview-445--hugo-theme-bootstrap.netlify.app/

**Reply by ppipada** [reply]
The preview looks good 👍

**Comment by ppipada** [comment]
@razonyang I pulled the latest release. 
Observations:
- If we have `page = ["HTML", "RSS"]`, present in outputs in `config.toml`, the rss links in pages point to the page itself and not to categories or tags or even home. Not sure if that is expected.
- If the above config is commented and we have only outputs configured for "home", everything works as expected.

**Reply by razonyang** [reply]
@ppipada Yep, that is expected. What's the intention of generating RSS for pages?

**Reply by ppipada** [reply]
got it. No need for rss for pages. Was just asking if this is considered as error scenario and handled accordingly, or left as is.
